### PR TITLE
Add CSV template support

### PR DESCRIPTION
**Pull request by sishbi**
Fixes issue #826 
1. Add csv.tpl to contrib directory
2. Add 'replaceAll' template function used by csv.tpl to replace commas with semi-colons for issue description
   to avoid unnecessary columns in CSV output

Signed-off-by: SimonBillingsley <sishbi@mac.com>

### CODE REVIEW

File: pkg/report/writer_test.go [hunk 1]
```diff
@@ -317,6 +317,12 @@ func TestReportWriter_Template(t *testing.T) {
 			template:      `{{ toLower (getEnv "AWS_ACCOUNT_ID") }} {{ getCurrentTime }}`,
 			expected:      `123456789012 2020-08-10T07:28:17.000958601Z`,
 		},
+		{
+			name:          "happy path: replace colon with dash",
+			detectedVulns: []types.DetectedVulnerability{},
+			template:      `{{- $time := getCurrentTime -}} {{ toLower (getEnv "AWS_ACCOUNT_ID") }} {{ replaceAll $time ":" "-" }}`,
```
Comment (knqyf263) [inline pkg/report/writer_test.go:323]: Any reason to output `AWS_ACCOUNT_ID`? Looks like this test is irrelevant to AWS.

```suggestion
			template:      `{{- $time := getCurrentTime -}} {{ replaceAll $time ":" "-" }}`,
```

File: pkg/report/writer.go [hunk 2]
```diff
@@ -204,6 +204,9 @@ func NewTemplateWriter(output io.Writer, outputTemplate string) (*TemplateWriter
 		"escapeString": func(input string) string {
 			return html.EscapeString(input)
 		},
+		"replaceAll": func(input string, old string, new string) string {
```
Comment (knqyf263) [inline pkg/report/writer.go:207]: After merging https://github.com/aquasecurity/trivy/pull/832, you can use `replace`.
http://masterminds.github.io/sprig/strings.html

It is about to be merged. Just a moment, please.

Comment (knqyf263) [reply pkg/report/writer.go:207]: #832 got merged. Could you rewrite this function by `spring`?
http://masterminds.github.io/sprig/

Comment (sishbi) [reply pkg/report/writer.go:207]: If 'sprig' functions are available to templates then this PR isn't required, and I can use 'replace' instead.

**Comment by sishbi** [comment]
Closing as sprig 'replace' can do what this PR was asking for...

**Comment by knqyf263** [comment]
I thought `csv.tpl` was still worth adding even after `replaceAll` got to be unnecessary.

File: pkg/report/writer.go (hunk 2 above)
Comment (dcomas) [reply pkg/report/writer.go:207]: > If 'sprig' functions are available to templates then this PR isn't required, and I can use 'replace' instead.

@sishbi are you able to provide an example how you made it in the end? Interested to use CSV output as well. thanks.

**Comment by sishbi** [comment]
Reopen PR to use replace function from sprig for csv.tpl

File: pkg/report/writer.go (hunk 2 above)
Comment (sishbi) [reply pkg/report/writer.go]: @dcomas I have reopened the PR and used the 'replace' function from sprig for the csv.tpl

**Comment by sishbi** [comment]
@knqyf263 I have reopened the PR and committed the fixes required to get csv.tpl into trivy

**Comment by knqyf263** [comment]
@sishbi I'm sorry for the late reply. It looks like the merge doesn't seem to work and there are a lot of differences. Could you look into it?

**Comment by sishbi** [comment]
Replaced by #919

**Pull request closed without merging**